lib/ui/screens/search/search_controller_impl.dart

```diff
@@ -32,16 +29,32 @@ class SearchControllerImpl extends _$SearchControllerImpl
   }
 
   @override
-  void onSearch({required String query}) {
+  void onSearch({required String query}) async {
```

Comment:
**suggestion:** We should think about if we even need the results at this point in time in the model. If this method is called on every update we wont need the results form the model anywhere, we can just take the live data.
Also, only set the state once, because each setter triggers an update. 

A little overkill for this application but in general we should debounce this call so there will only be a call if there is no new update for some milliseconds to reduce traffic.

Reply:
tell me how or do it by ur self xd